embedded/src/main/java/org/wildfly/core/embedded/Configuration.java

```diff
@@ -0,0 +1,432 @@
+            }
+            return new Builder(jbossHome);
+        }
+
+        /**
+         * Creates a new builder for the configuration.
+         *
+         * @param jbossHome the servers home directory
+         */
+        public static Builder of(final Path jbossHome) {
+            if (jbossHome == null) {
+                throw EmbeddedLogger.ROOT_LOGGER.nullVar("jbossHome");
+            }
+            return new Builder(jbossHome);
+        }
+
+        /**
+         * Adds a command argument.
+         *
+         * @param arg the argument to add, if {@code null} the argument is ignored
+         *
+         * @return this builder
+         */
+        public Builder addCommandArgument(final String arg) {
+            if (arg != null) {
+                cmdArgs.add(arg);
+            }
+            return this;
+        }
+
+        /**
+         * Adds the arguments to the command line arguments,
+         *
+         * @param args the arguments to add, if {@code null} the argument is ignored
+         *
+         * @return this builder
+         */
+        public Builder addCommandArguments(final String... args) {
+            if (args != null) {
+                cmdArgs.addAll(Arrays.asList(args));
+            }
+            return this;
+        }
+
+        /**
+         * Sets the command line arguments replacing any previously set arguments.
+         *
+         * @param args the arguments to set, if {@code null} the arguments are cleared
+         *
+         * @return this builder
+         */
+        public Builder setCommandArguments(final String... args) {
+            cmdArgs.clear();
+            return addCommandArguments(args);
+        }
+
+        /**
+         * Sets a hint for the JBoss Logging facade on which log manager the loggers so delegate to.
+         * <p>
+         * Sets the {@code org.jboss.logging.provider} system property and adds the hinted logging module to the list
+         * of system packages.
+         * </p>
+         *
+         * @param loggerHint the logger hint
+         *
+         * @return this builder
+         */
+        public Builder setLoggerHint(final LoggerHint loggerHint) {
+            this.loggerHint = loggerHint;
+            return this;
+        }
+
+        /**
+         * Sets a hint for the logging subsystem to indicate whether or not subsystem should configure logging on a
+         * special log context or not. A value of {@code true} indicates the special log context should be used. A
+         * value of {@code false} indicates no special log context should be used. The default is {@code true}.
+         *
+         * @param embeddedLoggingHint {@code false} if the logging subsystem should not use a special log context for
+         *                            it's configuration
+         *
+         * @return this builder
+         */
+        public Builder setEmbeddedLogContextHint(final boolean embeddedLoggingHint) {
```

Comment:
Is this basically "setUseEmbeddedLogContext(boolean)?  What's the difference between a hint and directly stating what to do?

Reply:
Essentially this was done for offline CLI. Maybe I should just remove this method and explicitly set the `org.wildfly.logging.embedded` system property in offline CLI.